frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -452,19 +452,25 @@ func dispatchWorkdir(d *dispatchState, c *instructions.WorkdirCommand, commit bo
 }
 
 func dispatchCopy(d *dispatchState, c instructions.SourcesAndDest, sourceState llb.State, isAddCommand bool, cmdToPrint interface{}, chown string) error {
+	for _, src := range c.Sources() {
```

Comment:
I guess we need to modify `copy` cmd like `copy --unpack=0,2` (only unpacks the 1st arg and the 3rd arg).

As we are going to remove the `copy` cmd soon, can we get this merged as workaround?

Reply:
@AkihiroSuda Are you sure these combinations are supported/documented? I would guess that if you want the unpack case you should never specify more than one source path. Or at least the behavior of others is undefined. That limitation seems better for a temp solution as well than messing with history.

Reply:
Docker seems actually accepting `ADD https://example.com/remote.tar.gz local.tar.gz /foo`  (unpacks `local.tar.gz` but keeps `remote.tar.gz` as-is), although not sure it is documented.

Maybe we should modify Docker to print warning for such `ADD` instruction?

Reply:
@AkihiroSuda Yes, we can add a warning but I don't think we should document it (or try to change the behavior unless it makes implementation simpler). It should be an undefined case and therefore users are not recommended to use it.

Reply:
Bit unrelated but maybe we should support an array of `meta` in executor. So multiple commands can be run for single layer. Shouldn't block this one but seems useful if anything similar shows up.

Reply:
Do we still support mixing up remote non-archives and local non-archives?
https://github.com/moby/moby/blob/8974fd47c7677b21c43597709655136addadb864/integration-cli/docker_cli_build_test.go#L535

Reply:
Updated PR to keep the history.

Mixing up remote archives and local ones doesn't print warning atm because we can't detect archives statically.
Solver should print warning/error when CopyOp is implemented.

@tonistiigi BTW is https://github.com/tonistiigi/copy/pull/2 mergeable?

Reply:
> BTW is tonistiigi/copy#2 mergeable?

I'll rebase tomorrow. Should be ok then.